core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -470,32 +469,68 @@ class SpendReportingService(
     bytesProcessedCounter += stats.getEstimatedBytesProcessed
   }
 
+  def getOwnedWorkspaceGoogleProjects(
+    childContext: RawlsRequestContext
+  ): Future[Map[RawlsBillingProjectName, Seq[Workspace]]] =
+    samDAO
+      .listResourcesWithActions(
+        SamResourceTypeNames.workspace,
+        SamWorkspaceActions.readSpendReport,
+        childContext
+      )
+      .flatMap {
+        case ownerWorkspaces if ownerWorkspaces.isEmpty =>
+          Future.successful(Map.empty[RawlsBillingProjectName, Seq[Workspace]])
+        case ownerWorkspaces =>
+          val validWorkspaceIds = ownerWorkspaces
+            .map(_.getResourceId)
+            .filter(resourceId => Try(UUID.fromString(resourceId)).isSuccess) // filter out non-UUIDs
+            .toList
+          workspaceServiceConstructor(childContext).getGCPWorkspacesByBillingProjects(validWorkspaceIds)
+      }
+
+  def getOwnedWorkspaceGoogleProjectsInProject(
+    projectName: RawlsBillingProjectName,
+    childContext: RawlsRequestContext
+  ): Future[Map[GoogleProjectId, WorkspaceName]] =
+    requireProjectAction(projectName, SamBillingProjectActions.createWorkspace) {
+      for {
+        // Get user owned workspaces in the project
+        projectNames <- getOwnedWorkspaceGoogleProjects(childContext).map {
+          _.flatMap { case (_, value) =>
+            value.collect {
+              case workspace if workspace.namespace == projectName.value =>
+                workspace.googleProjectId -> workspace.toWorkspaceName
+            }
+          }.toMap
+        }
+      } yield projectNames
+    }
+
   def getSpendForGCPBillingProject(
     project: RawlsBillingProjectName,
     start: DateTime,
     end: DateTime,
     aggregations: Set[SpendReportingAggregationKeyWithSub]
-  ): Future[SpendReportingResults] = {
+  ): Future[SpendReportingResults] = traceFutureWithParent("getSpendForGCPBillingProject", ctx) { childContext =>
     validateReportParameters(start, end)
-    requireProjectAction(project, SamBillingProjectActions.readSpendReport) {
-      for {
-        spendExportConf <- getSpendExportConfiguration(project)
-        projectNames <- getWorkspaceGoogleProjects(project)
+    for {
+      spendExportConf <- getSpendExportConfiguration(project)
+      projectNames <- getOwnedWorkspaceGoogleProjectsInProject(project, childContext)
```

Comment:
`getWorkspaceGoogleProjects` is now unused except in a test - clean it up?

Reply:
Fixed: https://github.com/broadinstitute/rawls/pull/3193/commits/802413ba4a91edce2ca73e664deee052fad7bd79